### ISSUE TITLE
Hackage hls-call-hierarchy-plugin-1.1.0.0 incompatible with hls-1.9

### ISSUE THREAD

**Issue by juhp** [haskell/haskell-language-server#3427]
I know the Build from Source instructions say to use git, but just checking whether you plan to upload all the latest deps to Hackage?
Building from Hackage worked for 1.8 and earlier, but maybe you don't intend to support that way anymore?
(In which case I will switch my Fedora HLS build to use git instead, though using Hackage is more reproducible.
Personally, I feel if one day we can get HLS into Stackage that would be even better, but that is probably still far.)

**Comment by wz1000** [comment]
I did upload to hackage:
https://hackage.haskell.org/package/haskell-language-server-1.9.0.0

What did you find missing?

**Comment by maralorn** [comment]
I think everything is fine besides a missing update for hls-call-hierarchy-plugin. The released hls-call-hierarchy-plugin is not compatible with hls 1.9.

**Comment by juhp** [comment]
Okay I think maralorn is right - I was hitting some unrelated local packaging changes I had made for 1.8...

**Comment by juhp** [comment]
Specifically Hackage hls-call-hierarchy-plugin-1.1.0.0 requires hls-plugin-api < 1.6

**Comment by spacekitteh** [comment]
Yes, please upload the latest version of the plugin to hackage.

**Comment by spacekitteh** [comment]
@wz1000 
> I did upload to hackage: https://hackage.haskell.org/package/haskell-language-server-1.9.0.0 What did you find missing?


https://hackage.haskell.org/package/haskell-language-server-1.9.0.0/reports/2

**Comment by wz1000** [comment]
Published https://hackage.haskell.org/package/hls-call-hierarchy-plugin-1.2.0.0 and revised HLS (https://hackage.haskell.org/package/haskell-language-server-1.9.0.0/revisions/)

**Comment by cumber** [comment]
@wz1000  Did that revision to depend on `hls-call-hierarchy-plugin ^>= 1.2` ever make it into the actual source? It seems to have reverted to `^>= 1.1` in the hackage release of HLS 1.9.1

**Comment by tbidne** [comment]
> @wz1000 Did that revision to depend on `hls-call-hierarchy-plugin ^>= 1.2` ever make it into the actual source? It seems to have reverted to `^>= 1.1` in the hackage release of HLS 1.9.1

It is in [HEAD](https://github.com/haskell/haskell-language-server/blob/master/haskell-language-server.cabal#L231) but apparently did not make it into the hackage release of 1.9.1:

```cabal
build-depends: hls-call-hierarchy-plugin ^>= 1.1
```
https://hackage.haskell.org/package/haskell-language-server-1.9.1.0/src/haskell-language-server.cabal

You can see the builds [failed](https://hackage.haskell.org/package/haskell-language-server-1.9.1.0/reports/) with the expected `rejecting: hls-call-hierarchy-plugin-1.2.0.0` line.

Perhaps a hackage revision could fix this?

**Comment by cumber** [comment]
@tbidne Yes, I'm aware of that because [such a revision was made](https://hackage.haskell.org/package/haskell-language-server-1.9.0.0/revisions/) to HLS 1.9.0.0.

But then 1.9.1.0 was pushed to Hackage with the old version bounds, so it doesn't work again. We need another revision.

**Comment by juhp** [comment]
https://download.copr.fedorainfracloud.org/results/petersen/haskell-language-server/fedora-rawhide-x86_64/05634904-haskell-language-server/

```
[__2] next goal: hls-call-hierarchy-plugin (dependency of
haskell-language-server +callhierarchy)
[__2] rejecting: hls-call-hierarchy-plugin-1.2.0.0 (conflict:
haskell-language-server +callhierarchy => hls-call-hierarchy-plugin^>=1.1)
[__2] trying: hls-call-hierarchy-plugin-1.1.0.0
[__3] next goal: hls-plugin-api (dependency of haskell-language-server)
[__3] rejecting: hls-plugin-api-1.6.0.0 (conflict: hls-call-hierarchy-plugin
=> hls-plugin-api^>=1.5)
[__3] rejecting: hls-plugin-api-1.5.0.0 (conflict: haskell-language-server =>
hls-plugin-api^>=1.6)
```
It's not enabled in CI or should it be disabled by default?

**Comment by michaelpj** [comment]
I published a revision to 1.9.1.0. Revisions are tricky!
